src/Utility/Text.php

```diff
@@ -232,7 +232,7 @@ public static function insert(string $str, array $data, array $options = []): st
 
         foreach ($tempData as $key => $hashVal) {
             $key = sprintf($format, preg_quote($key, '/'));
-            $str = preg_replace($key, $hashVal, $str);
+            $str = preg_replace($key, (string)$hashVal, $str);
```

Comment:
If $tmpData is type array<string, string> what type is getting passed in?

Reply:
Apparently `$tempData` isn't exactly `array<string, string>` :slightly_smiling_face: 

https://travis-ci.org/github/ADmad/cakephp/jobs/710436188#L498

Reply:
I updated the inline docblock for `$tempData`.

Reply:
I see, it's from the call to `crc32` above. Since `crc32` will enforce string type anyway I suppose this is fine to cast here instead.